Cockpit 239 (#383)
cockpit-project/cockpit-project.github.io: _posts/2021-03-03-cockpit-239.md

---
title: Cockpit 239
author: lis
date: 2021-03-03
tags: cockpit linux
slug: cockpit-239
category: release

> summary: Cockpit with restyled updates page

Reviewer: This needs updating.

---

Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).  We release regularly.

Here are the release notes from Cockpit version 239.

> ### Machines: Allow editing disk cache mode

Reviewer 2: Probably one of these headline adjustments

```suggestion
### Machines: Disk cache mode editing
```

```suggestion
### Machines: Disk cache mode is now editable
```

Author: I prefer to have the headings be a verb describing what changed.

It's now possible to edit the cache mode for existing virtual machine

> images.  This was previously only possible when creating new images.  In

Reviewer: s/images/disks

Reviewer 3: I'd still include the screenshot from https://github.com/cockpit-project/cockpit/pull/15430: 

![screenshot](https://user-images.githubusercontent.com/42733240/109371195-22dfe400-78a4-11eb-8a3c-5bc82ca82e84.png)

That is well done and makes the blog post less of a wall of text.

Reviewer 2: Wow. Why is the storage volume name so high up? I guess it's in a tag that makes it monospaced, but it's still _so_ wrong.

Reviewer 2: Filed @ https://github.com/cockpit-project/cockpit/issues/15459

> images.  This was previously only possible when creating new images.  In

Reviewer: 'creating new disks'. Images are associated with the disk which contains an OS, right?

Author: I'm good with "disks".  thanks.

order to edit the cache mode, any machines using the volume need to be

> powered off.

Reviewer 4: I would just do s/image/disk .
What is done with this functionality is related to attaching disks. `Creating new image` is a different action (the one done thru `qemu-img`). 
The difference is that when you are attaching and editing disk, you already have an existing image. Then you only attach/add this image to a VM as a disk (and there you can configure such things as a cache mode)

> ### Packaging: Removed ./configure options for distribution specific packages

Reviewer 2: ```suggestion
### Packaging: Removed ./configure options for distribution-specific packages
```

The phrase needs a hyphen in this usage.

> The previous --with-vdo-package, --with-nfs-client-package,

Reviewer 3: I'd drop "previous" here. (I realize that I wrote the original)

The options would look better as `code` formatting, as otherwise -- is turned into a long dash. I suggest to copy my original markdown, which also has two links which got lost here. (Edit the description in https://github.com/cockpit-project/cockpit/pull/15415 to get that)

Also, let's move this paragraph to the bottom, to have the UI changes together.

--with-appstream-config-packages, and --with-appstream-data-packages
options for ./configure got removed. This information is now contained
in the respective manifest files, for example for the Storage page. This
makes the manifest files portable across distributions. They can still
be adjusted by administrators through override.json files.

> ### Logs: add link to the services page

Reviewer 3: Capital "Add" for consistency

> ### Logs: add link to the services page

Reviewer 2: ```suggestion
### Logs: Link to related services
```

I'm using link as a verb here. :wink:

When viewing the details of a log entry, there's now a link directly to
the relevant service page, if applicable.

> ### SELinux: Restyle to fit with other pages

Reviewer 2: Multiple suggestions again:

```suggestion
### SELinux: Restyle to resemble other pages
```

```suggestion
### SELinux: Restyle to look like other pages
```

The SELinux page has been restyled to better match the style of other

> pages in Cockpit.

Reviewer 3: A screenshot could not hurt (not that important, of course). But the one in https://github.com/cockpit-project/cockpit/pull/15364 is too large and wide.

Reviewer 2: Oh cool! I somehow missed this. Looks great!

Author: I've made my own screenshot for this one

---
title: Cockpit 239
author: lis
date: 2021-03-03
tags: cockpit linux
slug: cockpit-239
category: release
summary: Cockpit with various improvements
---

Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).  We release regularly.

Here are the release notes from Cockpit version 239.

## Terminal: Support for changing the font size

Users can now change the font size in the emulator window on the
terminal page.

![New font size selector in the terminal](/images/terminal-font-size.png)

### Machines: Allow editing disk cache mode

> It's now possible to edit the cache mode for existing virtual machine
> disks.  This was previously only possible when creating new disks.  In
> order to edit the cache mode, any machines using the volume need to be
> powered off.

Reviewer 2: ```suggestion
Cache mode is now editable for existing virtual machine images, when a machine is in a "shut off" state. Previously, it could only be set within Cockpit while creating a new image.
```

Author: The opinion — that passive voice should be avoided — is had by me.

Reviewer 2: It's now possible to not say something other than "it's now possible to" all the time in the release notes.

```suggestion
Edit cache mode for existing virtual disks in this new release. Previous versions of Cockpit only allowed editing the cache mode during disk creation. To edit the cache mode, machines using the volume need to be "shut off".
```

There were a lot of issues with the existing paragraph, including "powered off", as we call it "shut off" in the UI, as does libvirt and everything that uses it.

Reviewer: 'machines using the volume' is not correct here. volume is reffering to libvirt's storage volume concept, however not all disks have a corresponding volume.

Reviewer 2: What's the requirement to ensure the disk's cache mode can be edited?

![Cache mode editing for machines](/images/edit-vm-cache-mode.png)

### Logs: Link to related services page

> When viewing the details of a log entry, there's now a link directly to
> the relevant service page, if applicable.

Reviewer 2: ```suggestion
Details pages for log entries now link to related services, when applicable.
```

![Log entry linking to service](/images/log-link-to-service.png)

### SELinux: Restyle to resemble other pages

The SELinux page has been restyled to better match the style of other
pages in Cockpit.

![SELinux page with the new style](/images/selinux-restyled.png)

### Packaging: Removed ./configure options for distribution specific packages

> The `--with-vdo-package`, `--with-nfs-client-package`,
> `--with-appstream-config-packages`, and `--with-appstream-data-packages`
> options for `./configure` got removed. This information is now contained
> in the respective manifest files, for example for the [Storage
> page](https://github.com/cockpit-project/cockpit/blob/master/pkg/storaged/manifest.json.in).
> This makes the manifest files portable across distributions. They can
> still be adjusted by administrators through [override.json
> files](https://cockpit-project.org/guide/latest/packages.html#package-replace).

Reviewer 2: ```suggestion
Several distribution-specific options for `./configure` were removed, including:

- `--with-vdo-package`
- `--with-nfs-client-package`
- `--with-appstream-config-packages`
- `--with-appstream-data-packages`

This information is now contained in respective manifest files (`storaged` and `apps`), making manifest files more portable across distributions.

These values can still be adjusted by administrators in `override.json` files.
```

Reviewer 2: I'm not sure about the "(`storaged` and `apps`)" part; please someone correct this if wrong. I do see the vdo/nfs align to Storage and Appstream would make sense for Applications.

---
title: Cockpit 239
author: lis
date: 2021-03-03
tags: cockpit linux
slug: cockpit-239
category: release
summary: Cockpit with various improvements
---

Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).  We release regularly.

Here are the release notes from Cockpit version 239.

## Terminal: Support for changing the font size

Users can now change the font size in the emulator window on the
terminal page.

![New font size selector in the terminal](/images/terminal-font-size.png)

### Machines: Allow editing disk cache mode

It's now possible to edit the cache mode for existing virtual machine
disks.  This was previously only possible when creating new disks.  In
order to edit the cache mode, any machines using the volume need to be
powered off.

![Cache mode editing for machines](/images/edit-vm-cache-mode.png)

### Logs: Link to related services page

The detailed view for log entries now links to related services, when
applicable.

![Log entry linking to service](/images/log-link-to-service.png)

### SELinux: Restyle to resemble other pages

> The SELinux page has been restyled to better match the style of other

Reviewer 2: ```suggestion
The SELinux page now matches the style of other pages in Cockpit.
```

Removes the "has been restyled" passive voice.

pages in Cockpit.

![SELinux page with the new style](/images/selinux-restyled.png)

### Packaging: Removed ./configure options for distribution-specific packages

> Several distribution-specific options for `./configure` were removed, including:

Reviewer 2: ```suggestion
The `./configure` script dropped several distribution-specific options:
```

- `--with-vdo-package`
- `--with-nfs-client-package`
- `--with-appstream-config-packages`
- `--with-appstream-data-packages`

> This information is now contained in respective manifest files, making manifest files more portable across distributions.

Reviewer 2: ```suggestion
Manifest files now contain this same information, which is more portable across distributions.
```

> These values can still be adjusted by administrators in `override.json` files.

Reviewer 2: ```suggestion
Administrators can still adjust these values in `override.json` files.
```
